CoffeeShop.BLL/Services/GenericCoffeeShopService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using AutoMapper;
+using CoffeeShop.BLL.Interfaces;
+using CoffeeShop.DAL.Interfaces;
+
+namespace CoffeeShop.BLL.Services
+{
+    public class GenericCoffeeShopService<TModel, TEntity> : IGenericCoffeeShopService<TModel> 
+        where TModel : class
```

Comment:
Just for your information, You can create interface for your models to make a restriction for your generic service. but this time it's ok